src/main/kotlin/com/github/markoresic/footballvotingdemo/controller/VoteController.kt

```diff
@@ -24,7 +26,7 @@ class VoteController(
     fun createVote(@PathVariable playerId: String): ResponseEntity<Unit> {
 
         if (!playerService.playerExistsById(playerId)) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build()
+            throw NoSuchElementException("Player not found!")
```

Comment:
As in other examples error is not API friendly and does not return correct code and response body:

here is an example of ErrorHandler with @ControllerAdvice annotation which you can register for you Controler.    

Where error is a simple object 

data class SimpleApiError(val code: Int, val message: String, val type: String? = null)